Breeding-Insight/bi-web
# [BI-1438] login redirect: redirect from nonexistent page after login

# Description
**Story:** https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?modal=detail&selectedIssue=BI-1438

Added a check after the user comes back from logging to to check if the path they are being sent to is valid. Sends them to the home page if it isn't valid. 

Note: The url will show the bad url before navigating to the actual url. There is no page flash though since the bad url is caught before entering the route. We would need to implement this check on the backend to not have that url shown, which I don't think is feasible. 

# Dependencies
biapi: develop

# Testing
1. Make sure you're not logged in
2. Go to `http://localhost:8080/noexist`
3. You should be sent to the home screen
4. Log in
5. You should be sent to the home screen after logging in. 
6. Try to go to a page that doesn't exist
7. You should see a "Page Does Not Exist" message. 
8. Log out. 
9. Repeat steps except this time navigate to a page that does exist and make sure the redirect takes you there. 


# Checklist:
